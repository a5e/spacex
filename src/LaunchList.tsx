import React from 'react';
import LaunchListItem from './LaunchListItem';
import ILaunch from './data/ILaunch';

interface LaunchListProps {
    launches: ILaunch[]
    handleSelect: (flightNumber: number) => void
    selectedLaunchNumber: number
}

export default function LaunchList(props:LaunchListProps){
    return (
        <section className="mission_list">
            <ul>
            {props.launches.map((launch:ILaunch) => {
                    return <LaunchListItem 
                        launch={launch}
                        key={launch.flight_number + launch.mission_name}
                        handleClick={props.handleSelect}
                        isSelected={launch.flight_number === props.selectedLaunchNumber}/>
                })
            }
            </ul>
        </section>
    );
}