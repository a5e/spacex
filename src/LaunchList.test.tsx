import React from 'react';
import ReactDOM from 'react-dom';
import LaunchList from './LaunchList';
import renderer from 'react-test-renderer';


const launches = [
    {
        "flight_number": 1,
        "mission_name": "FalconSat",
        "launch_year": "2006",
        "launch_date_utc": "2006-03-24T22:30:00.000Z",
        "mission_patch_small": "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
    }, {
        "flight_number": 2,
        "mission_name": "DemoSat",
        "launch_year": "2007",
        "launch_date_utc": "2007-03-21T01:10:00.000Z",
        "mission_patch_small": "https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
    }, {
        "flight_number": 3,
        "mission_name": "Trailblazer",
        "launch_year": "2008",
        "launch_date_utc": "2008-08-02T03:34:00.000Z",
        "mission_patch_small": "https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
    }
]

test('LaunchList should have 3 items', () => {
  const component = renderer.create(
    <LaunchList 
      launches = {launches}
      handleSelect = {
        () => {}
      }
      selectedLaunchNumber={2}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Launch number 3 should be selected', () => {
    const component = renderer.create(
        <LaunchList 
          launches = {launches}
          handleSelect = {
            () => {}
          }
          selectedLaunchNumber={3}
        />
      );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});