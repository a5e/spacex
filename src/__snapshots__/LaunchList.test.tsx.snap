// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch number 3 should be selected 1`] = `
<section
  className="mission_list"
>
  <ul>
    <li
      className="mission_list_item"
      data-flight-number={1}
      onClick={[Function]}
    >
      <img
        alt="FalconSat mission's patch"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
       
      FalconSat
    </li>
    <li
      className="mission_list_item"
      data-flight-number={2}
      onClick={[Function]}
    >
      <img
        alt="DemoSat mission's patch"
        src="https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
      />
       
      DemoSat
    </li>
    <li
      className="mission_list_item selected"
      data-flight-number={3}
      onClick={[Function]}
    >
      <img
        alt="Trailblazer mission's patch"
        src="https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
      />
       
      Trailblazer
    </li>
  </ul>
</section>
`;

exports[`LaunchList should have 3 items 1`] = `
<section
  className="mission_list"
>
  <ul>
    <li
      className="mission_list_item"
      data-flight-number={1}
      onClick={[Function]}
    >
      <img
        alt="FalconSat mission's patch"
        src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
      />
       
      FalconSat
    </li>
    <li
      className="mission_list_item selected"
      data-flight-number={2}
      onClick={[Function]}
    >
      <img
        alt="DemoSat mission's patch"
        src="https://images2.imgbox.com/4f/e3/I0lkuJ2e_o.png"
      />
       
      DemoSat
    </li>
    <li
      className="mission_list_item"
      data-flight-number={3}
      onClick={[Function]}
    >
      <img
        alt="Trailblazer mission's patch"
        src="https://images2.imgbox.com/3d/86/cnu0pan8_o.png"
      />
       
      Trailblazer
    </li>
  </ul>
</section>
`;
